src/sys/expressionparser.cxx

```diff
@@ -77,12 +83,100 @@ class FieldT : public FieldGenerator {
   FieldGeneratorPtr clone(const list<FieldGeneratorPtr> UNUSED(args)) override {
     return std::make_shared<FieldT>();
   }
-  double generate(double UNUSED(x), double UNUSED(y), double UNUSED(z),
-                  double t) override {
-    return t;
+  double generate(const Context& ctx) override {
+    return ctx.t();
   }
   std::string str() const override { return std::string("t"); }
 };
+
+class FieldParam : public FieldGenerator {
+public:
+  FieldParam(const std::string name) : name(name) {}
+  double generate(const Context& ctx) override {
+    return ctx.get(name); // Get a parameter
+  }
+  std::string str() const override { return std::string("{") + name + std::string("}"); }
```

Comment:
For reference, we can now stick `using namespace std::string_literals;` at the top of the file, and then we get to use `""s` string literals:

```cpp
  std::string str() const override { return "{"s + name + "}"s; }
```

(not saying it's actually necessary to use though!)

Reply:
Excellent! Yes, I've now replaced all the `std::string("...")` in this file with `"..."s`